packages/kbn-palettes/classes/color_fn_palette.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { Optional } from 'utility-types';
+import { KbnBasePaletteConfig, KbnBasePalette } from './palette';
+import { IKbnPalette, KbnPaletteType } from './types';
+
+const DEFAULT_COLOR_COUNT = 10;
+
+export interface KbnColorFnPaletteConfig extends Optional<KbnBasePaletteConfig, 'colorCount'> {
+  type: KbnPaletteType;
+  colorFn: (n: number) => string[];
+  defaultNumberOfColors?: number;
```

Comment:
From the properties name is hard to understand  the difference between `colorCount` and `defaultNumberOfColors`.
Can you please make this more clear?
`colorCount` is described as "Number of colors to display" so probably, and I believe `defaultNumberOfColors`, which is only used by the elastic palette, is actual number of colors in the palette.
Can we find a better variable name for these?

Reply:
Yeah the naming of either of these is not very descriptive. I added a description for `defaultNumberOfColors` in 1dc23ab2e158b14810dd794640f02c826eb240a4.

The difference is that `colorCount` is only used for displaying colors, while `defaultNumberOfColors` is used when calling the `colors` method without a specified `n`. This use case is to get the default number of colors defined by the palette function but we need to override this for the `euiPaletteColorBlind`.

I think we can clean this up when eui cleans up the palette functions and parameters.